src/ray/raylet/node_manager.cc

```diff
@@ -414,43 +404,56 @@ void NodeManager::Heartbeat() {
           ResourceSet(local_resources.GetTotalResources()));
     }
 
+    if (!last_heartbeat_resources_.GetAvailableResources().IsEqual(
+            local_resources.GetAvailableResources())) {
+      if (local_resources.GetAvailableResources().IsEmpty()) {
+        (*heartbeat_data->mutable_resources_available())[kResourcesTurningEmpty] = 0.0;
+      } else {
+        for (const auto &resource_pair :
+             local_resources.GetAvailableResources().GetResourceMap()) {
+          (*heartbeat_data->mutable_resources_available())[resource_pair.first] =
+              resource_pair.second;
+        }
+      }
+      last_heartbeat_resources_.SetAvailableResources(
+          ResourceSet(local_resources.GetAvailableResources()));
```

Comment:
How about abstract a function?

Reply:
total / available / load resources are handled tiny differently. So a function might not be a proper way.